README.md

```diff
@@ -137,7 +137,74 @@ This now has created the following structure, either on disk or as an in memory
 
 ## API
 
-See https://ipfs.github.io/js-ipfs-repo
+### new Repo(path[, options])
+
+Creates an IPFS Repo.
+
+Arguments:
+
+* `path` (string, mandatory): the path for this repo
+* `options` (object, optional): may contain the following values
+  * `lock` (string, defaults to `"fs"` in Node.js, `"memory"` in the browser): what type of lock to use. Lock has to be acquired when opening.
+  * `storageBackends` (object, optional): may contain the following values, which should each be a class implementing the [datastore interface](https://github.com/ipfs/interface-datastore#readme):
+    * `root` (defaults to [`datastore-fs`](https://github.com/ipfs/js-datastore-fs#readme) in Node.js and [`datastore-level`](https://github.com/ipfs/js-datastore-level#readme) in the browser)
+    * `blocks` (defaults to [`datastore-fs`](https://github.com/ipfs/js-datastore-fs#readme) in Node.js and [`datastore-level`](https://github.com/ipfs/js-datastore-level#readme) in the browser)
+    * `datastore` (defaults to `datastore-level`)
+
+```js
+const repo = new Repo('path/to/repo')
+```
+
+### repo.init (callback)
+
+Creates the necesary folder structure inside the repo.
+
+### repo.open (callback)
+
+Locks the repo.
+
+### repo.close (callback)
+
+Unlocks the repo.
+
```

Comment:
Let's break the API description into 3 sets:

- Set up - init, open, close and exists
- Repos - Explaining that there are two repos
- Utility/Convinience - apiAddr, config, version..